labelbox_dev/data_row.py

```diff
@@ -90,98 +84,108 @@ def _format_data_rows(data_rows):
-        return DataRow(data_row_json)
+    def create(dataset_id,
+               data_rows: List[CreateDataRowInput]) -> List["DataRow"]:
+        DataRow._check_sync_request_limits('create', len(data_rows))
+        data_rows = DataRow._format_data_rows(data_rows)
+        body = {
+            'dataset_id': dataset_id,
+            'data_rows': data_rows,
+        }
 
-    @staticmethod
-    def get_by_ids(
-        data_row_ids,
-        run_async: bool = False
-    ) -> Union[Iterator["DataRow"], BulkGetDataRowsTask]:
-        body = {'keys': data_row_ids, 'run_async': run_async}
-
-        if run_async:
-            task_json = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkget",
-                                             json=body)
-            # TODO: change to snake_case from backend
-            task_id = task_json.get('taskId')
-
-            if task_id is None:
-                raise LabelboxError(
-                    f"Failed to retrieve task information for `data_row.get_by_ids()` async operation"
-                )
-            return BulkGetDataRowsTask.get_by_id(task_id)
-
-        return IdentifierPaginator(f"{DATA_ROW_RESOURCE}", DataRow,
-                                   data_row_ids)
+        # TODO: Need to paginate sync call. Currently, 100 data rows is the limit
+        data_rows = Session.post_request(f"{DATA_ROW_RESOURCE}", json=body)
+        return [DataRow(data_row_json) for data_row_json in data_rows]
 
     @staticmethod
-    def get_by_global_keys(
-        global_keys,
-        run_async: bool = False
-    ) -> Union[Iterator["DataRow"], BulkGetDataRowsTask]:
-        body = {
-            'keys': global_keys,
-            'is_global_key': True,
-            'run_async': run_async
-        }
-        if run_async:
-            task_json = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkget",
-                                             json=body)
-            # TODO: change to snake_case from backend
-            task_id = task_json.get('taskId')
-
-            if task_id is None:
-                raise LabelboxError(
-                    f"Failed to retrieve task information for `data_row.get_by_global_keys()` async operation"
-                )
-            return BulkGetDataRowsTask.get_by_id(task_id)
+    def get_by_ids(ids: List[str],) -> Iterator["DataRow"]:
+        return IdentifierPaginator(f"{DATA_ROW_RESOURCE}", DataRow, ids)
 
+    @staticmethod
+    def get_by_global_keys(global_keys: List[str],) -> Iterator["DataRow"]:
         return IdentifierPaginator(f"{DATA_ROW_RESOURCE}",
                                    DataRow,
                                    global_keys,
                                    identifiers_key='global_keys')
+
+    @staticmethod
+    def update_async(data_rows: List[UpdateDataRowInput]) -> UpdateDataRowsTask:
+        for data_row in data_rows:
+            if 'row_data' in data_row:
+                data_row = DataRow._format_data_rows([data_row])[0]
```

Comment:
I wonder if `data_rows` actually gets updated, my interpretation is that this would only update the pointer to the local data_row variable but not the original data_rows object. Do we need `data_row.update(DataRow._format_data_rows([data_row])[0])`?

Same for other uses for the `update` method.

Reply:
Tested it and it works to update the `data_rows` var